libexec/python/cli.py

```diff
@@ -77,6 +76,14 @@ def main():
                         help="boolean to specify that the CMD should be included as a runscript (default is not included)", 
                         default=False)
 
+
+    # Flag to disable cache
+    parser.add_argument("--no-cache", 
+                        dest='disable_cache', 
+                        action="store_true",
+                        help="boolean to specify disabling the cache.", 
+                        default=False)
+
```

Comment:
ok - here is something to discuss! This command basically means that the layers that are downloaded are removed after running (with shutil.rmtree(cache_base)). This is different from the scenario when a user doesn't specify a cache, and then it goes to /tmp (and the docker portion doesn't clean it up). If the post python portion cleans up the files by removing the directory, then we are ok. If not, or if there is doubt, I think the base software should set this --no-cache variable on the getgo, that way the python always is in charge of making sure the files are cleaned up. 

In the current implementation, If the pre-docker portion always exports SOMETHING as SINGULARITY_CACHEDIR, this command isn't useful, because it would find that variable and use /tmp. However, given that this env variable is defined, this means that those files aren't cleaned up after.  So it comes down to this question:

- Does the user not specifying an environmental variable for the cache mean he/she wants to save the layers in /tmp? If so, we leave as is
- Does the user not specifying a cache mean he/she would prefer them to be pulled and then deleted? If so, we probably want the calling functions to add `--no-cache`.